File: src/C#/Pillont.Shodo.Domain/UseCases/Weather/Get/GetWeatherHandler.cs
using MediatR;
using Pillont.Shodo.Domain.UseCases.Weather.Get.Ports;
using Pillont.Shodo.Domain.UseCases.Weather.Models;

namespace Pillont.Shodo.Domain.UseCases.Weather.Get;

public class GetWeatherHandler : IRequestHandler<GetWeatherCommand, WeatherModel>
{
    private IWeatherProvider Provider { get; }

    public GetWeatherHandler(
        IWeatherProvider provider)
    {
        Provider = provider;
    }

    public Task<WeatherModel> Handle(GetWeatherCommand request, CancellationToken cancellationToken)
    {
        return Provider.GetAsync();
    }
}

File: src/Python/modules/domain/user_cases/weather/get/get_weather_handler.py
from typing import Final, override

from diator.requests import RequestHandler

from modules.domain.user_cases.weather.get.get_weather_command import GetWeatherCommand
from modules.domain.user_cases.weather.get.ports.iweather_provider import (
    IWeatherProvider,
)
from modules.domain.user_cases.weather.weater_model import WeatherModel


class GetWeatherHandler(RequestHandler[GetWeatherCommand, WeatherModel]):
    def __init__(self, provider: IWeatherProvider) -> None:
        self.provider: Final = provider

    @override
    async def handle(self, request: GetWeatherCommand) -> WeatherModel:
        return await self.provider.get_weater()
